test/e2e/integreatly_test.go

```diff
@@ -117,34 +102,70 @@ func integreatlyMonitoringTest(t *testing.T, f *framework.Framework, ctx *framew
 		return fmt.Errorf("failed to exec to pod: %s", err)
 	}
 
-	var promApiCallOutput Output
+	var promApiCallOutput apiResponse
 	err = json.Unmarshal([]byte(output), &promApiCallOutput)
 	if err != nil {
 		t.Logf("Failed to unmarshall json: %s", err)
 	}
+	var alertsResult prometheusv1.AlertsResult
+	err = json.Unmarshal(promApiCallOutput.Data, &alertsResult)
+	if err != nil {
+		t.Logf("Failed to unmarshall json: %s", err)
+	}
 
 	// Check if any alerts other than DeadMansSwitch are firing or pending
 	var firingalerts []string
 	var pendingalerts []string
 	var deadmanswitchfiring = false
-	var intlyalertpresent = false
-	for a := 0; a < len(promApiCallOutput.Data.Alerts); a++ {
-		if promApiCallOutput.Data.Alerts[a].Labels.Alertname == "DeadMansSwitch" && promApiCallOutput.Data.Alerts[a].State == "firing" {
+	for _, alert := range alertsResult.Alerts {
+		if alert.Labels["alertname"] == "DeadMansSwitch" && alert.State == "firing" {
 			deadmanswitchfiring = true
 		}
-		if promApiCallOutput.Data.Alerts[a].Labels.Alertname != "DeadMansSwitch" {
-			// ESPodCount will always fail since that alert is for OSD
-			// so it shouldn't cause our test to fail
-			if promApiCallOutput.Data.Alerts[a].Labels.Alertname == "KubePodCrashLooping" {
-				// Record that at least one integreatly alert is present
-				intlyalertpresent = true
+		if alert.Labels["alertname"] != "DeadMansSwitch" {
+			if alert.Labels["alertname"] == "KubePodCrashLooping" {
 				continue
 			}
-			if promApiCallOutput.Data.Alerts[a].State == "firing" {
-				firingalerts = append(firingalerts, promApiCallOutput.Data.Alerts[a].Labels.Alertname)
+			if alert.State == "firing" {
+				firingalerts = append(firingalerts, string(alert.Labels["alertname"]))
 			}
-			if promApiCallOutput.Data.Alerts[a].State == "pending" {
-				pendingalerts = append(pendingalerts, promApiCallOutput.Data.Alerts[a].Labels.Alertname)
+			if alert.State == "pending" {
+				pendingalerts = append(pendingalerts, string(alert.Labels["alertname"]))
+			}
```

Comment:
@matskiv Will pending alerts return in the `/alerts` endpoint?

Reply:
@davidffrench I don't know. I was curious about this as well, but I decided to leave it as is and get us unblocked ASAP. 
And I deleted my installation, so I can't check it now :(

Reply:
thanks @matskiv . Would be good to know this.